### ISSUE TITLE
bug(datepicker): Missing to announce the previous selection state on date, month and year

### ISSUE THREAD

**Issue by zarend** [angular/components#24086]
#### Reproduction

Steps to reproduce:
1. Open the [Datepicker Examples](https://material.angular.io/components/datepicker/examples)
2. Activate "Open calendar" button from any section, say, "Single date selection"
3. Apply any date by selecting one from the datepicker
4. Activate "Open the calendar" button again
5. Focus on the previously selected date value
 
 
#### Expected Behavior
It should announce the previously selected date as "selected" when focus on it

#### Actual Behavior
It is not announcing the previously selected value state by the screen reader

#### Environment
OS| Browser | Screen reader
ChromeOS | Google Chrome 91.0.4472.167 | ChromeVox
macOS Catalina | Google Chrome 92.0.4515.131 | VoiceOver

##### Additional Notes
This is copied from an [internal bug report](http://b/195600393) found during a11y testing.

**Comment by zarend** [comment]
This was fixed by #/24171
